packages/algob/src/lib/tx.ts

```diff
@@ -149,31 +186,80 @@ async function sendAndWait (
 export async function executeTransaction (
   deployer: Deployer,
   execParams: rtypes.ExecParams | rtypes.ExecParams[]): Promise<algosdk.ConfirmedTxInfo> {
-  const suggestedParams = await getSuggestedParams(deployer.algodClient);
-  const mkTx = async (p: rtypes.ExecParams): Promise<Transaction> =>
-    mkTransaction(p,
-      await mkTxParams(deployer.algodClient, p.payFlags, Object.assign({}, suggestedParams)));
-
-  let signedTxn;
-  if (Array.isArray(execParams)) {
-    if (execParams.length > 16) { throw new Error("Maximum size of an atomic transfer group is 16"); }
-
-    let txns = [];
-    for (const t of execParams) { txns.push(await mkTx(t)); }
-    txns = algosdk.assignGroupID(txns);
-    signedTxn = txns.map((txn: Transaction, index: number) => {
-      const signed = signTransaction(txn, execParams[index]);
-      deployer.log(`Signed transaction ${index}`, signed);
-      return signed;
-    });
-  } else {
-    const txn = await mkTx(execParams);
-    signedTxn = signTransaction(txn, execParams);
-    deployer.log(`Signed transaction:`, signedTxn);
+  try {
+    let signedTxn;
+    let txns: Transaction[] = [];
+    const txIDMap = new Map<number, string>();
+    if (Array.isArray(execParams)) {
+      if (execParams.length > 16) { throw new Error("Maximum size of an atomic transfer group is 16"); }
+
+      for (const [index, t] of execParams.entries()) {
+        txns.push(await mkTx(deployer, t, index, txIDMap));
+      }
+
+      txns = algosdk.assignGroupID(txns);
+      signedTxn = txns.map((txn: Transaction, index: number) => {
+        const signed = signTransaction(txn, execParams[index]);
+        deployer.log(`Signed transaction ${index}`, signed);
+        return signed;
+      });
+    } else {
+      const txn = await mkTx(deployer, execParams, 0, txIDMap);
+      signedTxn = signTransaction(txn, execParams);
+      deployer.log(`Signed transaction:`, signedTxn);
+      txns = [txn];
```

Comment:
If we're already treating a single txn as a group of 1 tx, then why not do `execParams = [execParams]` above (i think we do this in runtime as well) & remove the else at line 206 ? We are already passing index 0 in line 207. 
Is it because of the *console.log* ?

Reply:
because of `txns = algosdk.assignGroupID(txns);`

Reply:
ok